### PR TITLE
Add radio group controls for enums (#989)

### PR DESCRIPTION
**Pull request by AlexandraBuzila**
Adds MaterialRadioGroupControl  and vanilla RadioGroupControl that renders enums as radio groups.

selection_242

### CODE REVIEW

File: packages/material/src/controls/MaterialRadioGroupControl.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            uischema,
+            schema,
+            visible
+        } = this.props;
+        const isValid = errors.length === 0;
+        const style: { [x: string]: any } = {};
+        if (!visible) {
+            style.display = 'none';
+        }
+        const inputLabelStyle: { [x: string]: any } = {
```
Comment (edgarmueller) [inline packages/material/src/controls/MaterialRadioGroupControl.tsx]: Do we need this label style?

File: packages/examples/src/rule.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export const schema = {
 };
 
 export const uischema = {
-  type: 'HorizontalLayout',
+  type: 'VerticalLayout',
```
Comment (edgarmueller) [inline packages/examples/src/rule.ts]: I'd rather a new, explicit example for demonstrating the radio button group and revert this change.

File: packages/material/src/controls/MaterialRadioGroupControl.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import Radio from '@material-ui/core/Radio';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import {
+    FormControl,
+    FormControlLabel,
+    FormHelperText,
+    FormLabel
+} from '@material-ui/core';
+
+export class MaterialRadioGroupControl extends Control<ControlProps, ControlState> {
```
Comment (edgarmueller) [inline packages/material/src/controls/MaterialRadioGroupControl.tsx]: Do we have a vanilla version of this?

File: packages/material/test/renderers/MaterialRadioGroupControl.test.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    const inputs: HTMLInputElement[] = TestUtils.scryRenderedDOMComponentsWithTag(tree, 'input');
+    const radioButtons = _.filter(
+      inputs,
+      i => i.type === 'radio'
+    );
+    expect(radioButtons.length).toBe(4);
+    let currentlyChecked = _.filter(radioButtons, radio => radio.checked);
+    expect(currentlyChecked.length).toBe(1);
+    expect(currentlyChecked[0].value).toBe('D');
```
Comment (edgarmueller) [inline packages/material/test/renderers/MaterialRadioGroupControl.test.tsx]: We tested this expectation in the test before this one. Perhaps remove the assertions here and rename the tests accordingly. This test should only whether the binding works as intended.

File: packages/material/src/controls/MaterialRadioGroupControl.tsx (hunk 3 above)
Comment (AlexandraBuzila) [reply packages/material/src/controls/MaterialRadioGroupControl.tsx]: Pushed new commit containing vanilla renderer as well.

File: packages/vanilla/src/controls/RadioGroupControl.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                className={classNames.wrapper}
+                hidden={!visible}
+                onFocus={this.onFocus}
+                onBlur={this.onBlur}
+            >
+                <label htmlFor={id} className={classNames.label} >
+                    {computeLabel(isPlainLabel(label) ? label : label.default, required)}
+                </label>
+
+                <form>
```
Comment (eneufeld) [inline packages/vanilla/src/controls/RadioGroupControl.tsx]: why wrap in a form?

File: packages/vanilla/src/controls/RadioGroupControl.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                    {
+                        options.map(optionValue =>
+                            (
+                                <div>
```
Comment (eneufeld) [inline packages/vanilla/src/controls/RadioGroupControl.tsx]: why wrap each input into an own div?

File: packages/vanilla/src/controls/RadioGroupControl.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    isPlainLabel,
+    mapDispatchToControlProps,
+    mapStateToControlProps,
+    resolveSchema
+} from '@jsonforms/core';
+import { connectToJsonForms, Control } from '@jsonforms/react';
+import { VanillaControlProps } from '../index';
+import { addVanillaControlProps } from '../util';
+
+export class RadioGroupControl extends Control<VanillaControlProps, ControlState> {
```
Comment (eneufeld) [inline packages/vanilla/src/controls/RadioGroupControl.tsx]: In my opinion this can be a field, at least in the way it is currently implemented

File: packages/vanilla/src/controls/RadioGroupControl.tsx (hunk 5 above)
Comment (AlexandraBuzila) [reply packages/vanilla/src/controls/RadioGroupControl.tsx]: I replaced it with a div, needed for styling.

File: packages/vanilla/src/controls/RadioGroupControl.tsx (hunk 6 above)
Comment (AlexandraBuzila) [reply packages/vanilla/src/controls/RadioGroupControl.tsx]: Because we need to return an input and a label.

File: packages/vanilla/src/controls/RadioGroupControl.tsx (hunk 7 above)
Comment (AlexandraBuzila) [reply packages/vanilla/src/controls/RadioGroupControl.tsx]: Discussed it with Edgar, we left it as a control for now.

**Comment by AlexandraBuzila** [comment]
> the FormLabel is much bigger then all other labels in the form.

It is the same in the material examples: the FormLabel is big and does not change size when a value is selected in the radio group.
https://material-ui.com/demos/selection-controls/#radio-buttons

> Shouldn't we display Material and Vanilla RadioGroup in the same way?
> Material is showing the options as row, and vanilla as column.

I fixed it in the new commit.
